#!/usr/bin/env python
PACKAGE = "delta_manipulator"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("v_max", double_t, 0, "maximum_movement_speed",    0.5, 0.0,   1.0)
gen.add("v_traj", double_t, 0, "trajectory_movement_speed",    0.1, 0.0,   1.0)

gen.add("r", double_t, 0, "trajectory_radius",    0.5 , 0.,   0.5)

mode_enum = gen.enum([gen.const("static", int_t, 0, "static position hold"),
                 gen.const("line_z", int_t, 1, "line trajectory in x"),
                 gen.const("line_y", int_t, 2, "line trajectory in y"),
                 gen.const("circle_xy", int_t, 3, "circular trajectory")],
                 "select trajectory type")

gen.add("mode", int_t, 0, "mode selector", 0, 0, 10, edit_method=mode_enum)

exit(gen.generate(PACKAGE, "delta_manipulator", "Joystick"))